include/oneapi/dpl/internal/dynamic_selection.h

```diff
@@ -0,0 +1,175 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+
+#ifndef _ONEDPL_INTERNAL_DYNAMIC_SELECTION_H
+#define _ONEDPL_INTERNAL_DYNAMIC_SELECTION_H
+
+#include <memory>
+#include <utility>
+#include <list>
+
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+//ds_properties
+
+  namespace property {
+    struct universe_t {
+      static constexpr bool is_property_v = true;
+      static constexpr bool can_report_v = false;
+    };
+    inline constexpr universe_t universe;
+
+    struct universe_size_t {
+      static constexpr bool is_property_v = true;
+      static constexpr bool can_report_v = false;
+    };
+    inline constexpr universe_size_t universe_size;
+
+    struct task_completion_t {
+      static constexpr bool is_property_v = true;
+      static constexpr bool can_report_v = true;
+    };
+    inline constexpr task_completion_t task_completion;
+
+    template<typename T, typename Property>
+    auto query(T& t, const Property& prop) {
+      return t.query(prop);
+    }
+
+    template<typename T, typename Property, typename Argument>
+    auto query(T&& t, const Property& prop, const Argument& arg) {
+      return t.query(prop, arg);
+    }
```

Comment:
I guess `&&` might have been added in response to my earlier comment (which challenged the difference between `query` and `report`) - but then this change is formal only (because, unlike in `report`, there is no forwarding) and is only done in one overload instead of two.

Reply:
Yes, this should be done in all overloads and forward should be used.